Modules/_winapi.c

```diff
@@ -772,23 +772,88 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject* sortenvironmentkey(PyObject *module, PyObject *item) {
+    Py_ssize_t n;
+    wchar_t* s = PyUnicode_AsWideCharString(item, &n);
+    if (s == NULL) {
+        return NULL;
+    }
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT, LCMAP_UPPERCASE, s);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    ""
+};
+
+static PyObject *
+normalize_environment(PyObject* environment) {
+    PyObject *result, *keys, *keyfunc, *sort, *args, *kwargs;
+
+    keys = PyMapping_Keys(environment);
+    if (keys == NULL) {
+        return NULL;
+    }
+
+    keyfunc = PyCFunction_New(&sortenvironmentkey_def, NULL);
+    sort = PyObject_GetAttrString(keys, "sort");
+    args = PyTuple_New(0);
+    kwargs = PyDict_New();
+    PyDict_SetItemString(kwargs, "key", keyfunc);
+    if (PyObject_Call(sort, args, kwargs) == NULL) {
+        goto error;
+    }
+
+    result = PyDict_New();
+
+    for (int i = 0; i<PyList_GET_SIZE(keys); i++) {
+        if (i < 1) {
+            continue;
```

Comment:
```suggestion
    for (int i = 1; i<PyList_GET_SIZE(keys); i++) {
```

Reply:
Thanks for review! I had some code changes, so this suggest is outdated.